web-app/js/smartR/heatmapView.js

```diff
@@ -32,6 +33,94 @@ HeatmapView = (function(){
         }
     };
 
+    var Observable = function Observable() {
```

Comment:
Could you refactor this all BioMarkersModel in new file and then inject it in heatmapView